sw/airborne/modules/lidar/tfmini_i2c.c

```diff
@@ -0,0 +1,252 @@
+
+#define DISTANCE_MODE_REG_ADDR 0x50
+#define DISTANCE_MODE_SHORT    0x00   // for 0.3-2m
+#define DISTANCE_MODE_MIDDLE   0x03   // for 0.5-5m
+#define DISTANCE_MODE_LONG     0x07   // for 1-12m
+
+#define DETECTION_PATTERN_REG_ADDR  0x51
+#define AUTOMATIC_DETECTION_PATTERN 0x00
+#define FIXED_DETECTION_PATTERN     0x01
+
+#define SETTING_OF_RANGE_LIMIT_REG_ADDR 0x55
+#define RANGE_LIMIT_DISABLED 0x00
+#define RANGE_LIMIT_ENABLED  0x01
+
+#define RANGE_OUTPUT_LIMIT_THRESHOLD_REG_ADDR_L 0x56
+#define RANGE_OUTPUT_LIMIT_THRESHOLD_REG_ADDR_H 0x57
+
+//Settable 0x0000~0xFFFF
+//Unit: mm12000(DEC)
+#define LOWER_LIMIT_OF_SIGNAL_STRENGTH_THRESHOLD_REG_ADDR_L 0x58
+#define LOWER_LIMIT_OF_SIGNAL_STRENGTH_THRESHOLD_REG_ADDR_H 0x59
+
+//settable 0x0000~0xFFFF
+//When Strength is lower than that value, the distance output is 0xFFFF, and is not credible to be used as flag.20(DEC)
+#define UPPER_LIMIT_OF_SIGNAL_STRENGTH_THRESHOLD_REG_ADDR_L 0x5A
+#define UPPER_LIMIT_OF_SIGNAL_STRENGTH_THRESHOLD_REG_ADDR_H 0x5B
+
+//Settable 0x0000~0xFFFF
+#define OUTPUT_VALUE_OF_SIGNAL_STRENGTH_THRESHOLD_AT_THE_HIGHEST_POINT_REG_ADDR_L 0x5C
+#define OUTPUT_VALUE_OF_SIGNAL_STRENGTH_THRESHOLD_AT_THE_HIGHEST_POINT_REG_ADDR_H 0x5D
+
+//Settable 0x0000~0xFFFF
+#define UNIT_OF_DISTANCE_REG_ADDR 0x66
+#define UNIT_OF_DISTANCE_MM   0x00 //output millimeter (mm)
+#define UNIT_OF_DISTANCE_CM   0x01 // outout centimeter (cm)
+
+#define SLAVE_ADDRESS_REG_ADDR 0x26 //settable 0x10~0x78  default 0x10
+
+#define TRIGGER_MODE_REG_ADDR 0x27
+#define USE_EXTERNAL_TRIGGER  0 // max100Hz  default 0
+#define EXTERNAL_TRIGGER_REG_ADDR 0x01 //set to 0x01  Command for one single measurement
+
+#define DEVICE_RESET_ADDR 0x70 //setable 0x02  //All settings are reset to the default(excluding slave address and trigger mode)
+
+/// Ranger offset value for what considered the distance should be zero, e.g. high landing gear to tarmac 60cm still could be considerd zero
+#ifndef TFMINI_I2C_OFFSET
+#define TFMINI_I2C_OFFSET 0.0f
+#endif
+
+/// The minimum range for the device to be able to measure
+#ifndef TFMINI_I2C_MIN_RANGE
+#define TFMINI_I2C_MIN_RANGE 0.3f //Default for TFMini is 30cm(old firmware?), for TFMini Plus 10cm, sensors cannot measure a range closer than that
+#endif
+
+/// The maximum range for the device to be able to measure
+#ifndef TFMINI_I2C_MAX_RANGE
+#define TFMINI_I2C_MAX_RANGE 6.0f //Reasonable default for it is maximum value for both inside and outside lighting conditions in day environment
+#endif
+
+struct TFMiniI2C tfmini_i2c;
+
+/**
+ * Send measured value and status information so it can be read back in e.g. log file for debugging
+ */
+static void tfmini_i2c_send_lidar(struct transport_tx *trans, struct link_device *dev)
+{
+  pprz_msg_send_LIDAR(trans, dev, AC_ID,
+                      &tfmini_i2c.dist,
+                      &tfmini_i2c.status,
+                      (uint8_t*)tfmini_i2c.trans.status);//No guarantee on value but for telemetry of no importance
```

Comment:
`tfmini_i2c.trans.status` should not be cast to a pointer. Create a uint8 variable first and then pass its address.

Reply:
indeed, thx for spotting my lazyness ;)